draft-ietf-mls-protocol.md

```diff
@@ -4352,71 +4352,75 @@ current state of the group.  In practice, however, there is a risk
 that two members will generate Commit messages simultaneously,
 based on the same state.
 
-When this happens, there is a need for the members of the group to
-deconflict the simultaneous Commit messages.  There are two
-general approaches:
-
-* Have the Delivery Service enforce a total order
-* Have a signal in the message that clients can use to break ties
-
-As long as Commit messages cannot be merged, there is a risk of
-starvation.  In a sufficiently busy group, a given member may never
-be able to send a Commit message, because he always loses to other
-members.  The degree to which this is a practical problem will depend
+The extent to which this is a problem, and the appropriate solution, depends on
+the design of the Delivery Service. Per the CAP theorem, there are two general
+classes of distributed system:
+
+* Consistent and Partition-tolerant, or CP, systems can provide a globally
+  consistent view of data but may stop working if there are network issues.
+* Available and Partition-tolerant, or AP, systems continue working despite
+  network issues but may return different views of data to different users.
+
+Strategies for sequencing messages in CP and AP systems are described in the
+next two sections. Implementations MUST only update their cryptographic state
+when they have a reasonable degree of certainty that their Commit was accepted.
```

Comment:
This whole discussion belongs in the architecture document and not the protocol document.